lib/puppet/application/agent.rb

```diff
@@ -78,6 +78,10 @@ def preinit
     options[:job_id] = arg
   end
 
+  option("--waitforlock") do |arg|
+    options[:waitforlock] = true
+  end
```

Comment:
Since `waitforlock` is a puppet setting, the application-level option isn't needed (since all puppet settings can be specified for all applications).